lib/winston/transports/file.js

```diff
@@ -105,6 +108,10 @@ module.exports = class File extends TransportStream {
       return true;
     }
 
+    if (this.lazy) {
+      this.open();
```

Comment:
How does this work with e.g. file rotation?  Maybe it's fine, but I think it would be good if we could add one more test case that makes sure file rotation still works as expected with `lazy: true`.

Reply:
Yes indeed, current tests of file rotation don't have this covered. I will add some tests.

Reply:
Yeah, the lazy option messes up with rotation and stress tests. I think that it's because they are assuming immediate file creation (assertions are constructed that way), not that the outcome is invalid.